did_resolver/src/traits/resolvable/mod.rs

```diff
@@ -11,11 +11,12 @@ use self::{resolution_options::DidResolutionOptions, resolution_output::DidResol
 
 #[async_trait]
 pub trait DidResolvable {
-    type ExtraFields: Default;
+    type ExtraFieldsService: Default;
+    type ExtraFieldsOptions: Default;
```

Comment:
Why is `Default` required here? You can already pass dummy ZST types like `()` or whatever else to "avoid" the additional fields. The `Default` seems like having a risk of having to implement it for the type you want to pass, which might create it in an unusable state, which you have to take care of or you end up with garbage fields.
Or am I missing something?

Reply:
I was also debating whether to get rid of it, thinking of the same argument. It's not there to "avoid" additional fields, but more for convenience. Speaking of `ExtraFieldsOptions`, I don't see the problem of unusable state as present for the current use-case, but in general it is there. I will remove the bound from `ExtraFieldsOptions`, removing it from `ExtraFieldsService` is out of scope of this PR.

Reply:
Wasn't it added in `ExtraFieldsService` in this PR, though? May I ask why is it outside of scope?

Reply:
It wasn't, as you can see in the diff.

Reply:
Ah, my bad, the type was renamed but the `Default` was there.